proposal/first-class-mixins.md

```diff
@@ -0,0 +1,197 @@
+# First Class Mixins: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/626))*
+
+This proposal promotes mixins to first-class values and adds members to the
+`sass:meta` module for manipulating them.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Types](#types)
+  * [Operations](#operations)
+    * [Equality](#equality)
+  * [Serialization](#serialization)
+* [Functions](#functions)
+  * [`meta.type-of()`](#metatype-of)
+  * [`meta.get-mixin()`](#metaget-mixin)
+  * [`meta.module-mixins()`](#metamodule-mixins)
+  * [`meta.accepts-content()`](#metaaccepts-content)
+* [Mixins](#mixins)
+  * [`meta.apply()`](#metaapply)
+## Background
+
+> This section is non-normative.
+
+Sass today has two callables: the function and the mixin. Functions in Sass are
+a first class SassScript value type; they can be assigned to variables and
+invoked dynamically.
+
+Mixins, however, are not first class values. This is a stumbling block many
+users (sass/sass#626, sass/sass#673, sass/sass#3328, sass/sass#3439, among
+others) of Sass run into.
+
+Promoting mixins to be first class values would resolve a number of commonly
+requested features while also rounding out the language and adding feature
+parity between Sass's two callables.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal promotes mixins to first-class values, giving Sass composability
+for both of its kinds of callables.
+
+Additionally, it provides ways to interact with mixin values similarly to how
+one would with function values:
+
+* `meta.get-function()` => `meta.get-mixin()`
+
+* `meta.module-functions()` => `meta.module-mixins()`
+
+* `meta.call()` => `meta.apply()`
+
+## Types
+
+This proposal promotes the [mixin value][] to a Sass value type.
+
+[mixin value]: ../spec/at-rules/mixin.md#mixin
+
+### Operations
+
+The only operation permitted for mixin objects is checking for equality. All
+other operations throw an error.
+
+#### Equality
+
+Two user-declared mixins are considered equal if they refer to the exact same
+mixin as declared in the source code.
+
+Builtin mixins -- i.e. those declared by the Sass standard library, and not declared directly in Sass -- are equal to only themselves.
```

Comment:
This is still potentially ambiguous. Does "equal only to itself" mean that separate calls to `meta.get-mixin()` return unequal values even if they refer to a mixin of the same name from the same module?